src/hotspot/share/utilities/concurrentHashTable.inline.hpp

```diff
@@ -1298,4 +1339,75 @@ inline bool ConcurrentHashTable<CONFIG, F>::
   return true;
 }
 
+template <typename CONFIG, MEMFLAGS F>
+class ConcurrentHashTable<CONFIG, F>::BucketsClaimer {
+  // Default size of _claim_size_log2
+  static const size_t DEFAULT_CLAIM_SIZE_LOG2 = 7;
+
+  ConcurrentHashTable<CONFIG, F>* _cht;
+  // The table is split into ranges, every increment is one range.
+  volatile size_t _next_to_claim;
+  size_t _claim_size_log2; // Log number of buckets in claimed range.
+  size_t _limit;      // Limit to number of claims
+
+  // If there is a paused resize, we also need to operate on the already resized items.
+  volatile size_t _next_to_claim_new_table;
+  size_t _claim_size_log2_new_table;
+  size_t _limit_new_table;
+
+public:
+  BucketsClaimer(ConcurrentHashTable<CONFIG, F>* cht) :
```

Comment:
Would it be possible to add a claim size parameter (with default value `DEFAULT_CLAIM_SIZE_LOG2`)?

Particularly I'm not convinced that the default value is good :)